### DISCUSSION TITLE
How to update a paragraph field after a query?

### DISCUSSION THREAD

**Discussion by murilobd** [Budibase/budibase#6808] [Help] [answered]
Imagine the situation: there's a button inside a table to delete a row using a REST API Query. Right after the query is executed, I want to show to the user the response from that query, but I don't see how to do it. Could someone help me out on this, please?
Below an image with the structure of the components:
![image](https://user-images.githubusercontent.com/4275011/179743387-7c49a26d-6d1c-4ec5-b653-fbc8da9391eb.png)

**Comment by R2bEEaton** [comment, marked as answer]
Here's one way to do this:

1. On your delete button, add an "Update State" step as shown.
![image](https://user-images.githubusercontent.com/34921506/179809239-6401bf15-0f6c-4c25-920d-9476b1337c68.png)

2. In your paragraph, set the text to `{{ State.ParargaphText }}`.

This should result in the paragraph being updated with the query result (which you can change however you want with Handlebars or JS). 

### More info:
States are useful for removing data from the parent-child flow. They are also useful for moving data between pages or storing client information when leaving and coming back (if persistent).
https://docs.budibase.com/docs/app-state

**Reply by murilobd** [reply]
That works! Thanks a lot @R2bEEaton ! I confess I was using the State for global things...
The single remark of this solution is: if use the same strategy in another view and in a previous view the State.ParagraphText was set, it will display the "old value".

Is there an action to reset the state when user navigates to another view?

**Reply by R2bEEaton** [reply]
Good question, here's a way you can get around this.

On the delete button you could update a second state "timeout" or something to `{{ date now "x" }}` which is the time in milliseconds since January 1st, 1970. Then on your display paragraph you could use conditions to hide itself if `{{ subtract ( date now "x" ) State.timeout }}` is more than `1` (1 millisecond). This expression would be calculated on screen loading so this won't affect a paragraph that is already shown. If timeout is set, it hides the paragraph until it is set again to the current time.

Attached is an example. Hope this helps!
[ParagraphState-export-1658422793428.txt](https://github.com/Budibase/budibase/files/9161297/ParagraphState-export-1658422793428.txt)

And of course if you want to use the same thing in another view, you can also set different states: `ParagraphText1`, `ParagraphText2`, such that this `timeout` thing might not even be needed depending on your uses.
